internal/cmd/kafka/command_topic_produce_onprem.go

```diff
@@ -115,20 +129,35 @@ func (c *command) produceOnPrem(cmd *cobra.Command, args []string) error {
 		_ = os.RemoveAll(dir)
 	}()
 
-	// Meta info contains magic byte and schema ID (4 bytes).
-	schemaCfg := &sr.RegisterSchemaConfigs{
-		Subject:     subject,
-		SchemaDir:   dir,
-		SchemaType:  serializationProvider.GetSchemaName(),
-		ValueFormat: valueFormat,
-		SchemaPath:  schema,
-		Refs:        refs,
+	keySchemaConfigs := &sr.RegisterSchemaConfigs{
+		Subject:    keySubject,
+		SchemaDir:  dir,
+		SchemaType: keySerializer.GetSchemaName(),
+		Format:     keyFormat,
+		SchemaPath: keySchema,
+		Refs:       refs,
 	}
-	metaInfo, referencePathMap, err := c.registerSchemaOnPrem(cmd, schemaCfg)
+	keyMetaInfo, keyReferencePathMap, err := c.registerSchemaOnPrem(cmd, keySchemaConfigs)
 	if err != nil {
 		return err
 	}
-	if err := serializationProvider.LoadSchema(schema, referencePathMap); err != nil {
+	if err := keySerializer.LoadSchema(keySchema, keyReferencePathMap); err != nil {
+		return err
+	}
+
+	valueSchemaConfigs := &sr.RegisterSchemaConfigs{
+		Subject:    valueSubject,
+		SchemaDir:  dir,
+		SchemaType: valueSerializer.GetSchemaName(),
+		Format:     valueFormat,
+		SchemaPath: schema,
+		Refs:       refs,
+	}
+	valueMetaInfo, referencePathMap, err := c.registerSchemaOnPrem(cmd, valueSchemaConfigs)
+	if err != nil {
+		return err
+	}
+	if err := valueSerializer.LoadSchema(schema, referencePathMap); err != nil {
```

Comment:
may refer this as "valueSchema"

Reply:
I was calling it "schema" to match `--schema`. In v4 I want to change this flag to `--value-schema` though!